examples/src/main/java/io/dapr/examples/workflows/README.md

```diff
@@ -654,6 +655,212 @@ Key Points:
+```
+
+3. **Start app3 worker (handles App3FinalizeActivity):**
+```sh
+dapr run --app-id app3 --resources-path ./components/workflows --dapr-grpc-port 50003 --log-level=debug -- java -jar target/dapr-java-sdk-examples-exec.jar io.dapr.examples.workflows.crossapp.App3Worker
+```
+
+4. **Run the workflow client:**
+```sh
+java -Djava.util.logging.ConsoleHandler.level=FINE -Dio.dapr.durabletask.level=FINE -jar target/dapr-java-sdk-examples-exec.jar io.dapr.examples.workflows.crossapp.CrossAppWorkflowClient "Hello World"
+```
+
+<!-- STEP
+name: Run Cross-App Pattern workflow
+match_order: none
+output_match_mode: substring
+expected_stdout_lines:
+  - "=== Starting Cross-App Workflow Client ==="
+  - "Input: Hello World"
+  - "Created DaprWorkflowClient successfully"
+  - "Attempting to start new workflow..."
+  - "Started a new cross-app workflow with instance ID:"
+  - "Waiting for workflow completion..."
+  - "Workflow instance with ID:"
+  - "completed with result: HELLO WORLD [TRANSFORMED BY APP2] [FINALIZED BY APP3]"
+  - "=== WORKFLOW STARTING ==="
+  - "Starting CrossAppWorkflow:"
+  - "Workflow name:"
+  - "Workflow instance ID:"
+  - "CrossAppWorkflow received input: Hello World"
+  - "Workflow input: Hello World"
+  - "Calling cross-app activity in 'app2'..."
+  - "About to call cross-app activity in app2..."
+  - "=== App2: TransformActivity called ==="
+  - "Input: Hello World"
+  - "Output: HELLO WORLD [TRANSFORMED BY APP2]"
+  - "Calling cross-app activity in 'app3'..."
+  - "About to call cross-app activity in app3..."
+  - "=== App3: FinalizeActivity called ==="
+  - "Input: HELLO WORLD [TRANSFORMED BY APP2]"
+  - "Output: HELLO WORLD [TRANSFORMED BY APP2] [FINALIZED BY APP3]"
+  - "Final cross-app activity result:"
+  - "CrossAppWorkflow finished with:"
+  - "=== WORKFLOW COMPLETING WITH:"
+background: true
+sleep: 60
+timeout_seconds: 60
+-->
+
+<!-- END_STEP -->
+
+**Expected Output:**
+
+The client will show:
+```text
+=== Starting Cross-App Workflow Client ===
+Input: Hello World
+Created DaprWorkflowClient successfully
+Attempting to start new workflow...
+Started a new cross-app workflow with instance ID: 001113f3-b9d9-438c-932a-a9a9b70b9460
+Waiting for workflow completion...
+Workflow instance with ID: 001113f3-b9d9-438c-932a-a9a9b70b9460 completed with result: HELLO WORLD [TRANSFORMED BY APP2] [FINALIZED BY APP3]
+```
+
+The workflow demonstrates:
+1. The workflow starts in the main worker (crossapp-worker)
+2. Calls an activity in 'app2' using cross-app functionality
+3. Calls an activity in 'app3' using cross-app functionality
+4. The workflow completes with the final result from all activities
+
+This pattern is particularly useful for:
+- Microservices architectures where activities are distributed across multiple services
+- Service mesh scenarios where different apps handle different types of activities
```

Comment:
@cicoyle Service Mesh scenarios? What does that mean?

Reply:
I think I just mean the first bullet- I meant the first bullet idea with the second one but its not clear and redundant, so Ill rm it